packages/osv-offline-db/src/lib/purl-helper.ts

```diff
@@ -0,0 +1,33 @@
+import type { Ecosystem } from './ecosystem';
+
+// https://github.com/google/osv.dev/blob/edacbd3c32b1d632bad8c5b506a14d0850b2e537/osv/purl_helpers.py#L18-L30
+const PURL_ECOSYSTEMS: Record<Ecosystem, string> = {
+  'crates.io': 'cargo',
+  Go: 'golang',
+  Hex: 'hex',
+  Maven: 'maven',
+  NuGet: 'nuget',
+  npm: 'npm',
+  Packagist: 'composer',
+  PyPI: 'pypi',
+  RubyGems: 'gem',
+} as const;
+
+function urlEncode(packageName: string): string {
+  const parts = packageName.split('/');
+  return parts.map(encodeURIComponent).join('/');
```

Comment:
Can you add a test to exercise `encodeURIComponent`?

Reply:
Already there:
https://github.com/renovatebot/osv-offline/blob/49b7591e6be87fcb7648554e3d343f20398b83b8/packages/osv-offline-db/src/lib/purl-helper.unit.spec.ts#L12

I also went over all databases but it seems npm is the only one needing this.